docs/userguide/constraints.md

```diff
@@ -222,10 +233,123 @@ DiscreteLinkedParametersConstraint(
+values for the ``affected_parameters`` (which can be ignored).
+
+(DDC)=
+```{note}
+BayBE requires that all dependencies are declared in a single 
+``DiscreteDependenciesConstraint``, creating a ``SearchSpace`` from multiple 
+``DiscreteDependenciesConstraint``'s will throw a validation error.
+```
+
+In the example below we mimic a situation where there are two switches and each switch 
+activates two other parameters that are only relevant if the first switch is "on" / the 
+second switch is on "right" respectively.
+```python
+from baybe.constraints import DiscreteDependenciesConstraint, SubSelectionCondition
+
+DiscreteDependenciesConstraint(
+    parameters=["Switch1", "Switch2"],  # the two parameters upon which others depend
+    conditions=[
+        SubSelectionCondition(selection=["on"]),     # values of Switch1 that activate the affected parameters
+        SubSelectionCondition(selection=["right"]),  # values of Switch2 that activate the affected parameters
+    ],
+    affected_parameters=[
+      ["Solvent", "Fraction"],  # parameters affected by "Switch1"
+      ["Frame1", "Frame2"]      # parameters affected by "Switch2"
+    ],
+)
+```
+
+An end to end example can be found [here](../../examples/Constraints_Discrete/dependency_constraints).
 
 ### ``DiscretePermutationInvarianceConstraint``
-Content coming soon...
+Permutation invariance is a property where combinations of values of multiple
+parameters do not depend on their order due to some symmetry in the experiment. 
+Suppose that we are creating a mixture which has up to three solvents, i.e. parameters 
+"Solvent 1", "Solvent 2", "Solvent 3".
+In this situation, all combinations from the following table would be equivalent, 
+hence the ``SearchSpace`` should effectively only contain one of them.
+
+|   | Solvent 1    | Solvent 2    | Solvent 3    |
+|---|--------------|--------------|--------------|
+| 1 | Substance 43 | Substance 3  | Substance 12 |
+| 2 | Substance 43 | Substance 12 | Substance 3  |
+| 3 | Substance 3  | Substance 12 | Substance 43 |
+| 4 | Substance 3  | Substance 43 | Substance 12 |
+| 5 | Substance 12 | Substance 43 | Substance 3  |
+| 6 | Substance 12 | Substance 3  | Substance 43 |
+
+```{note}
+Complex properties such as permutation invariance not only affect the search space, but 
+should ideally also constrain the surrogate model. For instance, the kernels in a 
+Gaussian process can be made permutation invariant to reflect this constraint, which 
+generally results in a better learning curve. Note, however, that not all surrogate 
+models provided by BayBE can easlily include such invariances. In these cases the 
+invariance is silently ignored, which will not be an issue, but just means these models 
+cannot benefit from a priori known constraints and invariances between parameters.
+```
+
+Let's add to the mixture example the fact that not only the choice of substance but also 
```

Comment:
With this addition, I can see how the example makes sense. What is the reason that you do not include this already when introducing the example?

Reply:
no particular reason, jsut built the example like this 

1) Lets start with situation A
2) (partial if any) code or example for situation A
3) In reality and additional to situation A we also need to consider B
4) code for A+B

instead of this

1) Explain Situation which needs A + B
2) Code for A+B together

Lookig at this example these two variants are not even that different because the text is very short so I wonder after reading the entire parapraph are you still unhappy with it?

Reply:
Nope, after re-reading I am happy now